### DISCUSSION TITLE
Is there a way to detect dark/light mode switch?

### DISCUSSION THREAD

**Discussion by Jasenkoo** [vuepress/core#415] [Q&A] [answered]
Tried watching prefers-color-scheme from this [answer](https://stackoverflow.com/a/57795495/8328193) but is not triggered. 

Looked up the docs and also couldn't find anything, so I was wondering is there a way do detect when user switch dark/light mode?

**Comment by meteorlxy** [comment, marked as answer]
```ts
import { useDarkMode } from '@vuepress/theme-default/lib/client'
import { watch } from 'vue'

export default {
  setup() {
    const isDarkMode = useDarkMode()
    watch(isDarkMode, () => {
      // ...
    })
  },
}
```

**Reply by Jasenkoo** [reply]
Thanks, it works

**Reply by ottopaulsen** [reply]
Does it still work in beta.51?

**Reply by Jasenkoo** [reply]
Not sure, switched to vitepress
